lua/entities/terminator_nextbot_zambiemechaelite/shared.lua

```diff
@@ -0,0 +1,422 @@
+                local arc = EffectData()
+                arc:SetOrigin( arcPos )
+                arc:SetMagnitude( 12 )
+                arc:SetScale( 5 )
+                util.Effect( "ElectricSpark", arc )
+            end )
+        end
+        
+        for _, ent in ipairs( ents.FindInSphere( pos, radius ) ) do
+            if not IsValid( ent ) then continue end
+            
+            local entPos = ent:GetPos()
+            local dir = ( entPos - pos ):GetNormalized()
+            local dist = entPos:Distance( pos )
+            local distFrac = 1 - ( dist / radius )
+            
+            if ent:Health() and ent:Health() > 0 then
+                local dmg = DamageInfo()
+                dmg:SetDamage( damage * distFrac )
+                dmg:SetAttacker( game.GetWorld() )
+                dmg:SetInflictor( game.GetWorld() )
+                dmg:SetDamageType( DMG_BLAST )
+                dmg:SetDamageForce( dir * 40000 * distFrac )
+                ent:TakeDamageInfo( dmg )
+            end
+            
+            if ent:IsPlayer() or ent:IsNPC() or ent:IsNextBot() then
+                ent:SetVelocity( dir * 3000 * distFrac + Vector( 0, 0, 1200 * distFrac ) )
+            elseif IsValid( ent:GetPhysicsObject() ) then
+                local phys = ent:GetPhysicsObject()
+                phys:ApplyForceCenter( dir * phys:GetMass() * 2500 * distFrac + Vector( 0, 0, phys:GetMass() * 1000 * distFrac ) )
+            end
+            
+            if dist < radius * 0.5 and ent:IsPlayer() then
+                local burnTime = math.min( 8 * distFrac, 4 )
+                if burnTime > 0.5 then
+                    ent:Ignite( burnTime )
+                end
+            end
+        end
+        
+        util.ScreenShake( pos, 50, 25, 5, radius * 3 )
+        
+        local sprite = EffectData()
+        sprite:SetOrigin( pos )
+        sprite:SetScale( 30 )
+        sprite:SetMagnitude( 8 )
+        util.Effect( "cball_explode", sprite )
+    end )
+end
+
+function ENT:AdditionalFootstep( _pos, _foot, _sound, volume, _filter )
+    local moveSpeed = self:GetVelocity():Length()
+    local speedFrac = moveSpeed / self.RunSpeed
+    
+    local snd = "npc/antlion_guard/foot_heavy" .. math.random( 1, 2 ) .. ".wav"
+    local lvl = 95
+    local pit = 60
+    
+    self:EmitSound( snd, lvl, pit, volume + 0.8, CHAN_STATIC )
+    
+    if math.random( 1, 100 ) < 50 then
+        self:EmitSound( "npc/scanner/scanner_nearmiss" .. math.random( 1, 2 ) .. ".wav", 75, 50, 0.3, CHAN_STATIC )
+    end
+    
+    if moveSpeed > self.WalkSpeed then
+        local shakeAmt = 2 + ( speedFrac * 4 )
+        util.ScreenShake( self:GetPos(), shakeAmt, 20, 0.2, 400 + ( moveSpeed * 0.5 ) )
+    end
+    
+    if speedFrac > 0.5 then
+        local dust = EffectData()
+        dust:SetOrigin( _pos )
+        dust:SetScale( 3 )
+        dust:SetMagnitude( 2 )
+        util.Effect( "ThumperDust", dust )
+    end
+    
+    return true
+end
```

Comment:
Footsteps do not feel right at all
Thankfully i made a generic "Perfect footsteps" system
https://github.com/StrawWagen/termhunter/blob/main/lua/entities/terminator_nextbot/footsteps.lua#L39-L55

You can check the elite necro for a working example, it uses perfect footsteps iirc